src/libraries/System.Drawing.Common/src/System/Drawing/Imaging/EncoderParameter.cs

```diff
@@ -91,7 +91,7 @@ public EncoderParameter(Encoder encoder, byte value)
 
             _parameterValueType = EncoderParameterValueType.ValueTypeByte;
             _numberOfValues = 1;
-            _parameterValue = Marshal.AllocHGlobal(Marshal.SizeOf(typeof(byte)));
+            _parameterValue = Marshal.AllocHGlobal(sizeof(byte));
 
             if (_parameterValue == IntPtr.Zero)
```

Comment:
`AllocHGlobal` never returns `IntPtr.Zero`. It throws `OutOfMemoryException` instead. You can delete all these `if (_parameterValue == IntPtr.Zero)` checks.